.github/workflows/deploy.yml

```diff
@@ -9,12 +9,13 @@ jobs:
   deploy-to-test-env:
     runs-on: ubuntu-latest
     environment: Test
-    name: deploy-to-test
+    name: Deploy to Test environment
     env:
       TOKEN: ${{ secrets.DIGITAL_OCEAN_TOKEN }}
       URL: ${{ vars.DEPLOYMENT_URL }}
     steps:
-      - uses: kachick/wait-other-jobs@v3.6.0
+      - name: Wait for lint, test, build, and other workflows to succeed
```

Comment:
I believe GitHub Actions have built in ways to have workflows trigger other workflows, so you don't need to use a 3rd party tool to accomplish this (and also I think this will give some UI sugar to boot).

Check out https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore

If it does what I am claiming (I could be wrong!) then I think this concept should probably also be applied to `build`.

Reply:
No, unfortunately `workflow_run` will trigger the workflow N times for N prior jobs, and those will trigger regardless of success of the previous job. This is why there are a plethora of 3rd party tools.

Reply:
Perhaps a better tack would be to unify the jobs in a single workflow such that `needs` could be used, I think that works as expected out of the box unlike `workflow_run`.